src/main/java/com/zepben/evolve/services/network/ResistanceReactance.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.services.network
+
+import com.zepben.evolve.cim.iec61968.assetinfo.PowerTransformerInfo
+import com.zepben.evolve.cim.iec61968.assetinfo.TransformerEndInfo
+import com.zepben.evolve.cim.iec61970.base.wires.PowerTransformerEnd
+import com.zepben.evolve.cim.iec61970.base.wires.TransformerStarImpedance
+
+/**
+ * @property r Positive sequence series resistance.
+ * @property r0 Zero sequence series resistance.
+ * @property x Positive sequence series reactance.
+ * @property x0 Zero sequence series reactance.
+ */
+data class ResistanceReactance(
+    val r: Double,
+    val r0: Double,
+    val x: Double,
+    val x0: Double
+) {
+
+    fun isComplete(): Boolean = !r.isNaN() && !r0.isNaN() && !x.isNaN() && !x0.isNaN()
+
+}
+
+private fun ResistanceReactance?.mergeIfIncomplete(toMerge: () -> ResistanceReactance?): ResistanceReactance? {
+    return when {
+        this == null -> toMerge()
+        isComplete() -> this
+        else -> toMerge()?.let { rr ->
+            ResistanceReactance(
+                r.takeIf { !it.isNaN() } ?: rr.r,
+                r0.takeIf { !it.isNaN() } ?: rr.r0,
+                x.takeIf { !it.isNaN() } ?: rr.x,
+                x0.takeIf { !it.isNaN() } ?: rr.x0
+            )
+        } ?: this
+    }
+}
+
+fun PowerTransformerEnd?.resistanceReactance(): ResistanceReactance =
```

Comment:
From memory we discussed this would be on `TransformerEnd` and then would use r, r0, x, x0 if it was an instance of `PowerTransformerEnd`. I also think this should return a `ResistanceReactance?` instead of an incomplete instance as all the other extension functions return null if `this` is null.